testgen/src/header.rs

```diff
@@ -64,6 +67,29 @@ where
     m_secs.serialize(serializer)
 }
 
+// Serialize and deserialize the `Option<AppHash>`, delegating to the `AppHash`
+// serialization/deserialization into/from hexstring.
+mod app_hash_serde {
+    use super::*;
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<AppHash>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let app_hash = tendermint::serializers::apphash::deserialize(deserializer)?;
+        Ok(Some(app_hash))
+    }
+
+    pub fn serialize<S>(value: &Option<AppHash>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match value {
+            Some(inner) => tendermint::serializers::apphash::serialize(inner, serializer),
+            None => serializer.serialize_none(),
```

Comment:
This is wrong, strictly speaking: you either serialize `Option<AppHash>` or `AppHash`, but a mix will not work with some deserializers. It happens to work with `serde_json` because `Some` is always serialized just as a non-`null` value.

I've got a pending branch to fix `serializers::nullable` which still does it as well.

Reply:
[904a2d6](https://github.com/informalsystems/tendermint-rs/pull/1344/commits/904a2d6881d4e0f7ba05fe5d31ee6018fb7fffc8)